packages/grafana-prometheus/src/querybuilder/components/MetricCombobox.tsx

```diff
@@ -171,3 +176,22 @@ const formatKeyValueStringsForLabelValuesQuery = (query: string, labelsFilters?:
 
   return formatPrometheusLabelFiltersToString(queryString, labelsFilters);
 };
+
+const getMectricComboboxStyles = () => ({
+  addaptToParentQueryEditor: css({
+    // Take metrics explorer button into account
+    maxWidth: `calc(100% - 32px)`,
+  }),
+  addaptToParentVariableEditor: css({
+    maxWidth: '100%',
+    display: 'flex',
+    '[class*="InlineFieldRow"]': {
+      '> div': {
+        'label + div': {
+          // Take label and the metrics explorer button into account
+          maxWidth: `calc(100% - (160px + 32px))`,
```

Comment:
Is there any way we could approach this in the InlineFieldRow component without calculating from hardcoded values?